zaza/openstack/charm_tests/octavia/tests.py

```diff
@@ -63,12 +65,12 @@ class LBAASv2Test(test_utils.OpenStackBaseTest):
     def setUpClass(cls):
         """Run class setup for running LBaaSv2 service tests."""
         super(LBAASv2Test, cls).setUpClass()
-        cls.keystone_client = openstack_utils.get_keystone_session_client(
-            cls.keystone_session)
-        cls.neutron_client = openstack_utils.get_neutron_session_client(
-            cls.keystone_session)
-        cls.octavia_client = openstack_utils.get_octavia_session_client(
-            cls.keystone_session)
+        cls.keystone_client = ObjectRetrier(
+            openstack_utils.get_keystone_session_client(cls.keystone_session))
+        cls.neutron_client = ObjectRetrier(
+            openstack_utils.get_neutron_session_client(cls.keystone_session))
+        cls.octavia_client = ObjectRetrier(
+            openstack_utils.get_octavia_session_client(cls.keystone_session))
```

Comment:
I am a bit surprised to see these at the setUpClass level. Are we seeing failures here?

Reply:
Ah, I may not have explained how it works properly.  The `__init__(...)` method does the wrapping of the object, but it's when then client is *used* that the retry pattern is activated.  e.g.

```python
# wrap a client that is used to make calls.
client = ObjectRetrier(some_object_instance)
...
# later
...
# This is where the retries are done if needed
client.do_call()
```

Reply:
This makes complete sense. I am not sure why I didn't see it at review time.